### PR TITLE
Fixes 22034: No check for duplicate @ManagedBean names

### LINKED ISSUE #22034: No check for duplicate @ManagedBean names
[//]: # "Please remove these comments"
[//]: # "Provide a general summary of the issue in the Title above"

## Environment Details
* GlassFish Version (and build number): 5.0 builds
* JDK version: JDK 8
* OS: Generic
* Database: Generic

----------

## Problem Description
[//]: # "Describe the bug in detail highlighting current behavior vs expected behavior"
[//]: # "State if the problem is easily reproducible or happens intermittently"
[//]: # "Include stack traces or command outputs"

The Managed Beans 1.0 Specification [1] as well as the Java EE 7 javadoc [1] clearly states that...
[[
Managed Bean names must be unique within a Java EE module.
]]

This should be reported when deploying the application. However, when two different pojo class (say TestBean, and TestAnotherBean),  are annotated as (javax.annotation.)ManagedBean(s) with the same name -- '@'ManagedBean("mybean"), no meaningful error is reported saying duplicate '@'ManagedBean exists / found in the same module. 

[1] https://jcp.org/en/jsr/detail?id=316
[2] http://docs.oracle.com/javaee/7/api/javax/annotation/ManagedBean.html#value--

## Steps to reproduce
[//]: # "Step by step instructions to reproduce the problem"
[//]: # "Provide sample code/application if relevant"
Declare two ManagedBeans with same name, and refer them from say a Servlet...
<pre>
import javax.annotation.ManagedBean;
...

@ManagedBean("mybean")
public class TestBean {
}

@ManagedBean("mybean")
public class TestAnotherBean {
}

...

public class DemoServlet extends HttpServlet {

   @Resource(lookup = "java:module/mybean")
   private TestBean testBean;

   @Resource(lookup = "java:module/mybean")
   private TestAnotherBean testAnotherBean;
...
}
</pre>

## Impact of Issue
[//]: # "How has this issue affected you? What are you trying to accomplish?"
This leads to unexpected exception when accessing the Servlet (or application)
<pre>
 java.lang.IllegalArgumentException: Can not set demo.TestAnotherBean field demo.DemoServlet.testAnotherBean to demo.__EJB31_Generated__TestBean__Intf____Bean__
</pre>
[//]: # "Please choose one each Type, Component, and Priority label"

### PR DESCRIPTION
**Pull request by rahul-rs-srivastava**
Fixes #22034

### CODE REVIEW

**Comment by rahul-rs-srivastava** [comment]
@glassfishrobot Run CI tests please

**Review by mskdeepak-oracle — COMMENTED** [review]
Looks Good. 
You might want to add a devtest - A negative test  checking that deployment fails in case a duplicate Managed Bean is present.

File: appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/BundleDescriptor.java [hunk 1]
```diff
@@ -218,8 +218,27 @@ public EntityManagerFactory getEntityManagerFactory(String unitName) {
...
-            managedBeans.add(desc);
-            desc.setBundle(this);
+            boolean duplicate = false;
+            //check for uniqueness of ManagedBean name, if defined
+            if (desc.isNamed()) {
+                for (ManagedBeanDescriptor managedBeanDescriptor : managedBeans) {
+                    if (managedBeanDescriptor.isNamed() && desc.getName().equals(managedBeanDescriptor.getName())) {
+                        //duplicate ManagedBean found
+                        duplicate = true;
+                        String msg = String.format("Error processing ManagedBean \"%s\" with Name \"%s\", "
```
Comment (yaminikb) [inline appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/BundleDescriptor.java]: Message should come from a resource bundle instead of being hard coded.

**Review by yaminikb — CHANGES_REQUESTED** [review]

File: appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/BundleDescriptor.java (hunk 1 above)
Comment (rahul-rs-srivastava) [inline appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/BundleDescriptor.java]: Added message to resource bundle

**Review by rahul-rs-srivastava — COMMENTED** [review]
Fixed

**Comment by rahul-rs-srivastava** [comment]
@glassfishrobot Run CI tests please

**Comment by rahul-rs-srivastava** [comment]
@glassfishrobot Run CI tests please

**Review by yaminikb — APPROVED** [review]
